rq/job.py

```diff
@@ -513,25 +522,31 @@ def save_meta(self):
         meta = dumps(self.meta)
         self.connection.hset(self.key, 'meta', meta)
 
-    def cancel(self):
+    def cancel(self, pipeline=None):
         """Cancels the given job, which will prevent the job from ever being
         ran (or inspected).
 
         This method merely exists as a high-level API call to cancel jobs
         without worrying about the internals required to implement job
         cancellation.
         """
+        # TODO: Dependent jobs are by default in a DEFERRED state. Should we
```

Comment:
@selwin What do you think about the CANCELED state?

Reply:
Yes, introducing a canceled state and a corresponding `CanceledJobRegistry` is a good idea. This way, we can keep track of canceled jobs.

This should be implemented in a separate PR though. Feel free to open an issue for this so we don't forget :)